File: ClientApp/src/crypto-charts/lib/IndicatorReference.ts
import { MarketReference } from "./MarketReference";

export class IndicatorReference {
    public marketRef: MarketReference;
    public indicatorName: string;
    public timeFrame: number;
    public settings: any;

    constructor(exchangeName: string, symbol: string, indicatorName: string, timeFrame: number, settings: any) {
        this.marketRef = new MarketReference(exchangeName, symbol);
        this.indicatorName = indicatorName;
        this.timeFrame = timeFrame;
        this.settings = settings;
    }
}

File: Hubs/Types/IndicatorReference.cs
using System;
using System.Dynamic;
using CryptoBot.Exchanges;
using CryptoBot.Indicators;
using MessagePack;

namespace CryptoBotUI.Hubs.Types
{
    [MessagePackObject]
    public class IndicatorReference
    {
        [Key("indicatorName")]
        public string IndicatorName;

        [Key("marketRef")]
        public MarketReference MarketRef;

        [Key("timeFrame")]
        public Int64 TimeFrame;

        [Key("settings")]
        public ExpandoObject Settings;

        public IndicatorReference() { }

        public IndicatorReference(Indicator indicator)
        {
            IndicatorName = indicator.Name;
            MarketRef     = new MarketReference(indicator.Market);
            TimeFrame     = indicator.TimeFrame;
            Settings      = indicator.Settings;
        }
    }
}